include/spdlog/details/registry.h

```diff
@@ -144,12 +144,15 @@ template <class Mutex> class registry_t
         _async_mode = false;
     }
 
-    static registry_t<Mutex>& instance()
+    static registry_t<Mutex>* instance()
```

Comment:
Because we need to get its intance from the application to be able to set the instance inside the library. But this should be a `std::shared_ptr` instead.

From the application:
```cpp
auto shared_registry = spdlog::details::registry::instance();
MyLib::setLoggerRegistry(shared_registry);
```

In the library:
```cpp
static void MyLib::setLoggerRegistry(std::shared_ptr<spdlog::details::registry> registry)
{
    spdlog::details::registry::set_registry(registry};
}
```